### PR TITLE
(SIMP-6238) Update to puppet-gitlab v3.0.2

### PR DESCRIPTION
**Pull request by lnemsick-simp**
- Switched from vshn/gitlab 1.13.3 to its replacement puppet/gitlab
  3.0.2 and reworked the manifests, as appropriate, to work with the
  newer module.
- Added manage_package and package_ensure parameters to simp_gitlab

SIMP-6284 #close
SIMP-6285 #close
SIMP-6238 #close

### CODE REVIEW

**Comment by lnemsick-simp** [comment]
:heavy_check_mark: https://gitlab.com/simp/pupmod-simp-simp_gitlab/pipelines/52428363

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -1,37 +1,51 @@
...
-          let(:facts){
-            os_facts.merge({
-            :gitlab_systemd => os_facts.fetch('init_systems',{}).include?('systemd'),
-          })}
+          let(:facts) { os_facts }
 
           context 'simp_gitlab class without any parameters' do
             let(:params) {{ }}
-            it_behaves_like 'a structured module'
+
```
[the remark this answers is not in the document]
Comment (lnemsick-simp) [reply spec/classes/init_spec.rb]: It was only used in one place.  So I thought the test was more clear with the example group expanded.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -1,37 +1,51 @@
...
+            it { is_expected.to compile.with_all_deps }
+            it { is_expected.to create_class('simp_gitlab') }
             it { is_expected.to contain_class('simp_gitlab').with_trusted_nets(['127.0.0.1/32']) }
+            it { is_expected.to contain_class('simp_gitlab') }
```
[the remark this answers is not in the document]
Comment (lnemsick-simp) [reply spec/classes/init_spec.rb]: Fixed per review comment

File: manifests/init.pp [hunk 3]
```diff
@@ -122,37 +123,52 @@
 # @param ldap_group_base
 #   EE only
 #
-# @author simp
+# @param manage_package
+#   Whether simp_gitlab will manage the gitlab-[ce,ee] package.
+#
+#   - Set to true if you want simp_gitlab will set the GitLab root password,
```
Comment (trevor-vaughan) [inline manifests/init.pp]:
```suggestion
#   - Set to true if you want simp_gitlab to set the GitLab root password,
```

File: README.md [hunk 4]
```diff
@@ -72,8 +72,8 @@ https://simp-project.atlassian.net/.
 ### What `simp_gitlab` affects
 
 This module is a profile that integrates Gitlab with SIMP.  It makes extensive
-use of the component module [`vshn/gitlab`][vshn_gitlab].  It is important to
-note that `vshn-gitlab` configures the GitLab Omnibus's `/etc/gitlab/gitlab.rb`
+use of the component module [`puppet/gitlab`][puppet_gitlab].  It is important to
+note that `puppet-gitlab` configures the GitLab Omnibus's `/etc/gitlab/gitlab.rb`
 and runs `gitlab-ctl reconfigure`.
```
Comment (LongLiveCHIEF) [inline README.md]: `puppet-gitlab` no longer executes `gitlab-ctl reconfigure` explicitly unless you configure it to do so. Your package manager will do this whenever you install or upgrade the omnibus-gitlab package, as that's part of the omnibus's `post-install` script.

This same script in the omnibus also starts the gitlab supervisors for the relevant roles defined in the nodes `gitlab.rb`, and as such the `puppet_gitlab` module no longer manages the service explicitly by default.

I left some links to relevant info in the issue you opened in our repo. Let me know if you need any other information.

**Comment by LongLiveCHIEF** [comment]
This might be some interesting reading for you as well: https://github.com/voxpupuli/puppet-gitlab/issues/240

**Pull request merged**